scripts/test_screeps_official_deploy.py

```
@@ -0,0 +1,277 @@
+            activate_world=activate_world,
+            confirm=confirm,
+            repo_root=repo_root or artifact.parent,
+        )
+
+    def test_build_code_payload_uses_main_module(self) -> None:
+        payload = deploy.build_code_payload("main", "module text")
+
+        self.assertEqual(payload, {"branch": "main", "modules": {"main": "module text"}})
+
+    def test_redacts_secret_values_and_module_contents(self) -> None:
+        secret = "fixture-value"
```

Ruff (0.15.12)

[error] Possible hardcoded password assigned to: "secret"

(S105)